src/Data/Array.purs

```diff
@@ -4,7 +4,9 @@
 -- | `Data.Sequence` instead, which might give better performance for certain
 -- | use cases. This module is useful when integrating with JavaScript libraries
 -- | which use arrays, but immutable arrays are not a practical data structure
--- | for many use cases due to their poor asymptotics.
+-- | for many use cases due to their poor asymptotics. The running time of all
```

Comment:
Big O tells you the computational complexity, which is not quite the same as running time (although of course they are related). Additionally, "The running time" is singular so the verb should be "is" rather than "are" here. However, I think it might be easiest to sidestep these issues by writing "All array modification operations (...) are at least as slow as `O(n)` ...".